tests/nosetests/pyanaconda_tests/module_subscription_tests.py

```diff
@@ -219,6 +228,186 @@ def set_system_purpose_test(self):
         self.assertEqual(output_system_purpose_data.usage, "baz")
         self.assertEqual(output_system_purpose_data.addons, ["a", "b", "c"])
 
+    def subscription_request_type_validation_test(self):
+        """Test type validation for the SubscriptionRequest DBus structure works correctly."""
+
+        subscription_request = SubscriptionRequest()
+        # set password type
+        subscription_request.type = str(SubscriptionRequestType.USERNAME_PASSWORD.value)
+        self.assertEqual(subscription_request.type,
+                         str(SubscriptionRequestType.USERNAME_PASSWORD.value))
+
+        # set key type
+        subscription_request.type = str(SubscriptionRequestType.ORG_KEY.value)
+        self.assertEqual(subscription_request.type,
+                         str(SubscriptionRequestType.ORG_KEY.value))
+
+        # bogus type - should raise an exception
+        with self.assertRaises(InvalidValueError):
+            subscription_request.type = "type_that_does_not_exist"
+
+    def subscription_request_data_defaults_test(self):
+        """Test the SubscriptionRequest DBus structure defaults."""
+
+        # create empty SubscriptionRequest structure
+        empty_request = SubscriptionRequest()
+
+        # compare with expected default values
+        expected_default_dict = {
+            "type": get_variant(Str, str(DEFAULT_SUBSCRIPTION_REQUEST_TYPE.value)),
```

Comment:
This is not even correct. Please, always use constants when you set up the DBus structure.

Reply:
Enum dropped.